### DISCUSSION TITLE
How does multithreading with different websites work?

### DISCUSSION THREAD

**Discussion by jphfritsche** [seleniumbase/SeleniumBase#2409] [Q&A] [answered]
I am trying to navigate to two websites in multiple threads. 

```
import pytest
from seleniumbase import BaseCase

class TestWebsiteVisits(BaseCase):
    @pytest.mark.parametrize('url', [
        "https://nowsecure.nl/#relax","https://nowsecure.nl/"
    ])
    def test_visit_website(self, url):
        self.open(url)
        self.sleep(30) 
```

Ideally in a loop 

```
import subprocess
import time

def run_tests():
    subprocess.run(["pytest", "seleniumbase_test.py", "--uc", "-n", "2"])

if __name__ == "__main__":
    while True:
        print("testing")
        run_tests()
        print("sleeping")
        time.sleep(300)
```

My problem is related to this example https://github.com/seleniumbase/SeleniumBase/discussions/2241#discussioncomment-7487800 but as far as I understand this works only with a single website? 

Help would be very appreciated. Many thanks!

**Comment by mdmintz** [comment, marked as answer]
You have to use `parameterized` to pass in multiple args. (In your case, different websites.)

(`pytest.parametrize` doesn't work with `unittest.TestCase` inheritance.)

Here's an example of `parameterized` for multiple proxies:

```python
from parameterized import parameterized
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__, "-n3")

class ProxyTests(BaseCase):
    @parameterized.expand(
        [
            ["host1:port1"],
            ["host2:port2"],
            ["host3:port3"],
        ]
    )
    def test_multiple_proxies(self, proxy_string):
        self.get_new_driver(
            undetectable=True, proxy=proxy_string, multi_proxy=True
        )
        self.driver.get("https://browserleaks.com/webrtc")
        self.sleep(30)
```

![multi_webrtc_checks](https://github.com/seleniumbase/SeleniumBase/assets/6788579/9292cd57-f364-49a5-9cd3-3e81235a68ef)

**Reply by suredarv06** [reply]
This above code works fine, 

I have a situation to listen to a SQS queue and process that message with seleniumBase, the message will have URL ,proxy and the remaining attribute.
In the below code I will listen to queue indefinitely so the test will not end after processing each message . But I need generate test result after each message. But currently no result and log are printed Since test is running indefinitely.



**Is there any option available for that with multithreading?**

**Share your thoughts.**
Code snippet is attached below

```
from parameterized import parameterized
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__, "-n3")

class ProxyTests(BaseCase):
    @parameterized.expand(
        [
            ["host1:port1"],
            ["host2:port2"],
            ["host3:port3"],
        ]
    )
    def test_multiple_proxies(self, proxy_string):
        self.get_new_driver(
            undetectable=True, proxy=proxy_string, multi_proxy=True
        )
        while True:
               listenSqs(self)
        self.sleep(30)
```

**Reply by mdmintz** [reply]
Unless you're routing the traffic through a proxy that enables you to log the output, you'll need to break out of your infinite loop somehow so that your script can end and you can get SeleniumBase logs.
